Added X-Content-Type-Options (nosniff) security feature

SashaMcr opened the pull request:
A new 'nosniff' option was added to the security headers settings, enabling more configuration options for the X-Content-Type-Options header. This enhances security by preventing MIME type confusion attacks, ensuring that browsers interpret files correctly according to their declared content type.

Commit:
Added X-Content-Type-Options (nosniff) security feature

A new 'nosniff' option was added to the security headers settings, enabling more configuration options for the X-Content-Type-Options header. This enhances security by preventing MIME type confusion attacks, ensuring that browsers interpret files correctly according to their declared content type.

hissy commented on controllers/single_page/dashboard/system/environment/security_header_extended.php:
```diff
@@ -14,6 +14,7 @@ public function view()
         $this->set('coop', $this->getPackageConfig()->get('security.cross_origin_opener_policy'));
         $this->set('coep', $this->getPackageConfig()->get('security.cross_origin_embedder_policy'));
         $this->set('accessControlAllowOrigin', $this->getPackageConfig()->get('security.access_control_allow_origin'));
+        $this->set('nosniff', $this->getPackageConfig()->get('security.x_content_type_options_nosniff'));
```
@SashaMcr 

See other config keys. These do not include values.
It's `access_control_allow_origin` , not `access_control_allow_origin_none`
Please follow same naming rule

hissy commented on single_pages/dashboard/system/environment/security_header_extended.php:
```diff
@@ -91,6 +92,22 @@
+        <?= $form->label('', t('X-Content-Type-Options')) ?>
+        <div class="form-check">
+            <?= $form->checkbox('nosniff', '1', $nosniff) ?>
+            <?= $form->label('nosniff', t('Enable X-Content-Type-Options (nosniff)')) ?>
```
@SashaMcr 

nosniff is the value for x-content-type-options header. not an alternative name of the header.

hissy commented on src/Http/Middleware/ContentTypeOptionsMiddleware.php:
```diff
@@ -0,0 +1,35 @@
+        $response = $frame->next($request);
+
+        if ($response->headers->has('X-Content-Type-Options') === false) {
+            if ($this->config === true) {
```
@SashaMcr 

`nosniff` is the only allowed value for x-content-type-options header, but please use `nosniff` instead of `true` for the config value. it's better to follow the same rule of other config values.

Commit:
Refactor X-Content-Type-Options handling in security settings

This commit adjusts the treatment of the 'X-Content-Type-Options' security setting across several files in the package. The parameter value check has been enhanced to prevent empty string applications. The changes include updating configuration calls, modifying treatment in middleware, and adjusting form display.

SashaMcr commented:
@hissy 

はい、分かりました
修理しました

hissy commented:
LGTM!
